test/entity.spec.js

```diff
@@ -9,583 +9,648 @@ const CLIENT_MAIN_SRC_DIR = constants.CLIENT_MAIN_SRC_DIR;
+                    assert.file(expectedFiles.clientNg2WithRootFolder);
+                    assert.noFile(expectedFiles.gatling);
+                    assert.noFile(`${SERVER_MAIN_SRC_DIR}com/mycompany/myapp/web/rest/FooResource.java`);
+                });
             });
-            it('generates expected files', () => {
-                assert.file(`${CLIENT_MAIN_SRC_DIR}i18n/en/sampleMicroserviceBar.json`);
-                assert.file(expectedFiles.clientNg2GatewayMicroserviceEntity);
-                assert.noFile(expectedFiles.gatling);
-                assert.fileContent(`${CLIENT_MAIN_SRC_DIR}app/entities/sampleMicroservice/bar/bar.service.ts`, 'samplemicroservice');
-                assert.fileContent(
-                    `${CLIENT_MAIN_SRC_DIR}app/entities/sampleMicroservice/bar/bar.module.ts`,
-                    'SampleMicroserviceBarModule'
-                );
-                assert.noFile(`${SERVER_MAIN_SRC_DIR}com/mycompany/myapp/web/rest/BarResource.java`);
+
+            describe('with entity from mongodb microservice', () => {
+                before(done => {
+                    helpers
+                        .run(require.resolve('../generators/entity'))
+                        .inTmpDir(dir => {
+                            fse.copySync(path.join(__dirname, '../test/templates/default-gateway'), dir);
+                        })
+                        .withArguments(['baz'])
+                        .withPrompts({
+                            useMicroserviceJson: true,
+                            microservicePath: '../'
+                        })
+                        .on('end', done);
+                });
+
+                it('sets expected custom databaseType from the microservice', () => {
+                    assert.jsonFileContent('.jhipster/Baz.json', { databaseType: 'mongodb' });
+                });
+                it('generates expected files', () => {
+                    assert.file(expectedFiles.clientBazGatewayMicroserviceEntity);
+                });
+                it('generates a string id for the mongodb entity', () => {
+                    assert.fileContent(`${CLIENT_MAIN_SRC_DIR}app/shared/model/sampleMicroservice/baz.model.ts`, 'id?: string');
+                });
             });
         });
+    });
 
-        describe('with entity from microservice and custom client-root-folder', () => {
-            before(done => {
-                helpers
-                    .run(require.resolve('../generators/entity'))
-                    .inTmpDir(dir => {
-                        fse.copySync(path.join(__dirname, '../test/templates/default-gateway'), dir);
-                    })
-                    .withArguments(['foo'])
-                    .withPrompts({
-                        useMicroserviceJson: true,
-                        microservicePath: '../'
-                    })
-                    .on('end', done);
-            });
+    context('regeneration from json file', () => {
```

Comment:
Since I found no way to define an entity comment/javadoc using `entity` subgenerator by CLI, I added new tests that regenerate entities from `.jhipster/*.json` files.
All changes above are just introduction of a context `creation from CLI` and related tabulations changes.